site_preference_builder/customSitePrefBuilder.js

```diff
@@ -102,6 +103,16 @@ const buildCustomSitePrefObject = (sitePref) => {
   if (sitePref.defaultValue) {
     customSitePref['default-value'] = [sitePref.defaultValue];
   }
+  if (sitePref.setOfStringValues) {
```

Comment:
```suggestion
  if (sitePref.valueDefinitions) {
```

`set` prefix is commonly kept for `setter` methods.

you should keep your framework / platform naming convention and calls variables as it will used in the xml produced.

Here, `valueDefinitions` is probably a better choice than `setOfStringValues` (even if it is less generic ... we will see to factorise this in future if there is another use case ... for now, this variable is only used once, so KISS and keep in mind that probably YAGNI)

Reply:
setOfstrings is a [type in SFCC](https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/content/b2c_commerce/topics/active_merchandising/b2c_dynamic_group_attribute_types.html?resultof=%22%53%65%74%2d%6f%66%2d%53%74%72%69%6e%67%22%20), this option would only be used for theses attributes